lib/src/connector/course_connector.dart

```diff
@@ -101,38 +101,49 @@ class CourseConnector {
       //取得學期資料
       nodes = courseNodes[0].getElementsByTagName("td");
       SemesterJson semester = SemesterJson();
-      semester.year = nodes[1].text;
-      semester.semester = nodes[2].text;
+
+      // the title String of the first course table was stored seperately in its <td> element,
+      // but it currently stores all the information in a row,
+      // (ex: 學號：110310144　　姓名：xxx　　班級：電機三甲　　　 112 學年度 第 1 學期　上課時間表)
+      // so the RegExp is used to filter out only the number parts
+      final titleString = nodes[0].text;
+      final RegExp numberFilter = RegExp(r'\b\d+\b');
+      final Iterable<RegExpMatch> numberMatches = numberFilter.allMatches(titleString);
+      // the "numbers" should consist of three numerical values
+      // ex: [110310144, 112, 1]
+      final List<String> numbers = numberMatches.map((match) => match.group(0)).toList();
```

Comment:
```suggestion
      final List<String> studentSemesterDetails = numberMatches.map((match) => match.group(0)).toList();
```

How about using `studentSemesterDetails`? `numbers` is still unintuitive ?

Reply:
[ab8c839](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/ab8c839e6cae3f30e62e1fdb0845f23daafad480)
[8c9b6ab](https://github.com/NEO-TAT/tat_flutter/pull/214/commits/8c9b6abfa25c62da2b2949d0519ae26e9445b6b6)